### PR TITLE
Update all Readme links to 4.0.0-beta.2

### PR DESCRIPTION
**Pull request by backwind1233**
1. update readme links to 4.0.0-beta.2
2. remove condition in storage samples
3. update com.azure:azure-security-keyvault-jca to 2.2.0

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,67 +13,66 @@
...
-| cosmos           | [azure-spring-boot-starter-cosmos:3.9.0]                                                       | [spring-cloud-azure-data-cosmos-sample](cosmos/spring-cloud-azure-starter-data-cosmos/spring-cloud-azure-data-cosmos-sample)                                                                          |
-| cosmos           | [azure-spring-boot-starter-cosmos:3.9.0]                                                       | [spring-cloud-azure-cosmos-sample](cosmos/spring-cloud-azure-starter-cosmos/spring-cloud-azure-cosmos-sample)                                                                          |
+| cloudfoundry     | N/A                                                                                            | [azure-cloud-foundry-service-sample](cloudfoundry/azure-cloud-foundry-service-sample)                                                          |
+| cosmos           | [azure-spring-data-cosmos:3.13.1]                                                              | [cosmos-multi-database-multi-account](cosmos/azure-spring-data-cosmos/cosmos-multi-database-multi-account)                |
```
Comment (chenrujun) [inline README.md:31]: Why keep `azure-spring-data-cosmos` and delete `azure-security-keyvault-jca`?

File: storage/spring-cloud-azure-starter-storage-blob/storage-blob-sample/src/main/java/com/azure/spring/sample/storage/resource/BlobController.java [hunk 2]
```diff
@@ -23,7 +20,6 @@
  */
 @RestController
 @RequestMapping("blob")
-@ConditionalOnBean(AzureStorageBlobProtocolResolver.class)
```
Comment (chenrujun) [inline storage/spring-cloud-azure-starter-storage-blob/storage-blob-sample/src/main/java/com/azure/spring/sample/storage/resource/BlobController.java:26]: Why delete this?

File: storage/spring-cloud-azure-starter-storage-file-share/storage-file-sample/src/main/java/com/azure/spring/sample/storage/resource/FileController.java [hunk 3]
```diff
@@ -27,7 +24,6 @@
  */
 @RestController
 @RequestMapping("file")
-@ConditionalOnBean(AzureStorageFileProtocolResolver.class)
```
Comment (chenrujun) [inline storage/spring-cloud-azure-starter-storage-file-share/storage-file-sample/src/main/java/com/azure/spring/sample/storage/resource/FileController.java:30]: Same here.

File: README.md (hunk 1 above)
Comment (backwind1233) [reply README.md:31]: 1. this is azure-spring-data-cosmos, no released in 4.x
2. run-with-command-line-server-side doesn't depend on azure-security-keyvault-jca

File: storage/spring-cloud-azure-starter-storage-blob/storage-blob-sample/src/main/java/com/azure/spring/sample/storage/resource/BlobController.java (hunk 2 above)
Comment (backwind1233) [reply storage/spring-cloud-azure-starter-storage-blob/storage-blob-sample/src/main/java/com/azure/spring/sample/storage/resource/BlobController.java:26]: in this sample, not autowired this bean

File: storage/spring-cloud-azure-starter-storage-file-share/storage-file-sample/src/main/java/com/azure/spring/sample/storage/resource/FileController.java (hunk 3 above)
Comment (backwind1233) [reply storage/spring-cloud-azure-starter-storage-file-share/storage-file-sample/src/main/java/com/azure/spring/sample/storage/resource/FileController.java:30]: in this sample, not autowired this bean

**Review by stliu — APPROVED** [review]

**Pull request merged**